feat: Added sidebar component to community page

lancemccluskey opened the pull request:
<!--
Please read the [Code of Conduct](https://github.com/nodejs/nodejs.dev/blob/master/CODE_OF_CONDUCT.md) and the [Contributing Guidelines](https://github.com/nodejs/nodejs.dev/blob/master/CONTRIBUTING.md) before opening a pull request.
-->

## Description

<!-- Write a brief description of the changes introduced by this PR -->

Added sidebar component in `pages/community.tsx` based on figma design [here](https://www.figma.com/file/lOxAGGg5KXb6nwie7zXkz6/NJ---Design-System?node-id=1338%3A12737).

## Related Issues

Fixes #1145 
<!--
  Link to the issue that is fixed by this PR (if there is one)
  e.g. Fixes #1234, Addresses #1234, Related to #1234, etc.
-->

<!--
  If you want to generate a preview of this PR on our staging server please
  make a comment on the Pull-Request with the text `/preview`
 -->

lancemccluskey reviewed (commented):
\preview

mikeesto commented on src/pages/community.tsx:
```diff
@@ -1,23 +1,113 @@
+                      isLearn={false}
+                      isRead={false}
+                      isActive={false}
+                      onClick={(): void => console.log(`${title} clicked.`)}
```
```suggestion
```

Can we remove this?

mikeesto reviewed (commented):
Direct preview link: https://staging.nodejs.dev/1174/community/

Looks good! 👍

manishprivet commented:
Clicking on section links redirects to learn section. Is that the supposed behavior?

Also, each section redirects to a heading on the main community page. Can we try making different pages/MD files for each section?

lancemccluskey commented:
I dont think that was in scope for this ticket. This was just to make the sidebar component. I assumed those would be in a future issue. @benhalverson would have more insight though

benhalverson commented:
Yes thats right @lancemccluskey

designMoreWeb reviewed (approved):
LGTM

benhalverson commented:
Can you look into why we are getting a conflict @lancemccluskey

lancemccluskey commented:
Yeah I'll take a look

lancemccluskey commented:
@benhalverson @designMoreWeb I added a few tests since I added a prop to the `NavigationItem` component. Im not seeing any conflicts now Ben. Or were you referring to the code coverage stuff?

benhalverson commented:
![Screen Shot 2021-03-24 at 7 28 44 PM](https://user-images.githubusercontent.com/7907232/112409713-3fbadc00-8cd7-11eb-9285-1a9a6bda9a2a.png)
I see this error when I try to rebase and merge

lancemccluskey commented:
Ahhh okay that makes more sense now. Ill fix the conflicts

lancemccluskey commented:
Okay I following the directions for that issue in a stackoverflow I found [here](https://stackoverflow.com/questions/50012036/github-cant-rebase-my-feature-branch-this-branch-cannot-be-rebased-due-to-con) so hopefully its all good on your end.

benhalverson commented:
nope still the same message 😢

lancemccluskey commented:
Whoops I accidentally pulled the old commits into my branch instead of pushing the new stuff after rebasing. NOW everything should be fine lol. Do you think well ever switch to just using a clone of the repo instead of forks? I honestly dont understand the benefit of using forks
